src/components/dialogs/generator-modification-dialog.js

```diff
@@ -153,31 +160,68 @@ const GeneratorModificationDialog = ({
             : { id: '' };
     }, [formValues]);
 
-    const [generatorInfos, generatorIdField] = useAutocompleteField({
+    const [generatorId, generatorIdField] = useAutocompleteField({
         label: 'ID',
         validation: { isFieldRequired: true },
         inputForm: inputForm,
         formProps: filledTextField,
-        values: equipmentOptions?.sort(compareById),
+        values: equipmentOptions?.sort((a, b) => a.localeCompare(b)),
         allowNewValue: true,
         getLabel: getIdOrSelf,
         defaultValue:
-            equipmentOptions?.find((e) => e.id === formValueEquipmentId?.id) ||
+            equipmentOptions?.find((e) => e === formValueEquipmentId?.id) ||
             formValueEquipmentId,
         loading: loadingEquipmentOptions,
     });
 
+    const id = useMemo(() => {
+        let id;
+        if (typeof generatorId === 'object') {
+            if (generatorId?.id !== '') {
+                id = generatorId?.id;
+            }
+        } else {
+            if (generatorId !== '') {
+                id = generatorId;
+            }
+        }
+        return id ?? formValueEquipmentId?.id;
+    }, [generatorId, formValueEquipmentId]);
+
+    const fetchGeneratorInformation = async (
```

Comment:
I'm not sure we should start using async - await keyword, please use Promise.then() instead to homogenize the code base